docs/adr/adr_gcloud-high-availability.md

```diff
@@ -0,0 +1,72 @@
+
+> The purpose of an HA configuration is to reduce downtime when a zone or instance becomes unavailable. This might happen during a zonal outage, or when there's a hardware issue. With HA, your data continues to be available to client applications.
+> The HA configuration provides data redundancy. A Cloud SQL instance configured for HA is also called a regional instance and has a primary and secondary zone within the configured region. Within a regional instance, the configuration is made up of a primary instance and a standby instance. Through synchronous replication to each zone's persistent disk, all writes made to the primary instance are replicated to disks in both zones before a transaction is reported as committed. In the event of an instance or zone failure, the standby instance becomes the new primary instance. Users are then rerouted to the new primary instance. This process is called a failover.
+
+For the statviz project, we introduced a *read replica* of the primary database instance to transfer the load for reading large data sets (relatively to the data read for common app usage). The read replica resides in the `europe-west1-c` zone. Furthermore [2]:
+
+> In a disaster recovery scenario, you can promote a replica to convert it to a primary instance. This way, you can use it in place of an instance that's in a region that's having an outage. You can also promote a replica to replace an instance that's corrupted.
+> The read replica is an exact copy of the primary instance. Data and other changes on the primary instance are updated in almost real time on the read replica.
+> Read replicas are read-only; you cannot write to them. The read replica processes queries, read requests, and analytics traffic, thus reducing the load on the primary instance.
+
+A primary instance cannot failover to a read replica, and read replicas are unable to failover in any way during an outage. [3]
+
+Our read replica does not have HA mode enabled itself.
+
+Finally, we have nightly back-ups configured on the primary database instance with point-in-time recovery enabled.
+
+## Decision Drivers
+
+1. **Data safety**: we want user and app data to persist even if a GCloud outage happens
+1. **Expenses**: we're being billed extra for HA mode [4]. The read replica is charged the same rate as a standard SQL instance [5].
+1. **App availability**: we want to minimize app downtime if a GCloud outage happens
+
+## Considered Options
+
+A. Keep HA
+
+B. Turn off HA
+
+## Decision
+
+Turn off HA.
+
+This is performed by editing the primary database instance in the GCloud console [6].
+
+## Consequences
+
+1. *Data safety* In case of a GCloud outage, data is still available in the back-ups (which can be restored when the primary instance runs again but lost its data). Another possibility is to promote the read-replica in place of the failed primary instance.
+1. *Expenses* We have less monthly costs for using GCloud.
```

Comment:
@HaGuesto can you state how much turning off HA would save us on average per month?

Reply:
30 EUR per month currently.